layer/filestore.go

```diff
@@ -98,6 +100,14 @@ func (fm *fileMetadataTransaction) SetCacheID(cacheID string) error {
 	return ioutil.WriteFile(filepath.Join(fm.root, "cache-id"), []byte(cacheID), 0644)
 }
 
+func (fm *fileMetadataTransaction) SetForeignSource(ref *distribution.Descriptor) error {
```

Comment:
Basically cosmetic, but I think in most existing code `distribution.Descriptor` is passed by value instead of by reference.